File: chobit eclipse java/chobit/TimeGate.java
package chobit;

import java.util.Calendar;
import java.util.Date;

public class TimeGate {
	// a gate that only opens x minutes after it has been set
	private int pause = 1;
	private Date openedGate = addMinutesToJavaUtilDate(new Date(), pause);
	private Date checkPoint = new Date();
	public TimeGate(int minutes) {
		super();
		this.pause = minutes;
	}
	public TimeGate() {
	}
	public Boolean isClosed() {
		return !openedGate.before(new Date());
	}
	public void close() {
		this.openedGate = addMinutesToJavaUtilDate(new Date(), pause);
	}
	public void close(int minutes) {
		Date now = new Date();
		openedGate = addMinutesToJavaUtilDate(now, minutes);
	}
	private Date addMinutesToJavaUtilDate(Date date, int minutes) {
	    Calendar calendar = Calendar.getInstance();
	    calendar.setTime(date);
	    calendar.add(Calendar.MINUTE, minutes);
	    return calendar.getTime();
	}
	public void setPause(int pause) {
		if(pause <60 && pause >0) {
		this.pause = pause;}
	}

	public void resetCheckPoint() {
		this.checkPoint = new Date();
	}

	public int givenTwoDateTimesInJava8_whenDifferentiatingInSeconds_thenWeGetTen() {
		Date now = new Date();
		long diff = now.getTime() - this.checkPoint.getTime();
		long diffSeconds = diff / 1000 % 60;
		// long diffMinutes = diff / (60 * 1000) % 60;
		// long diffHours = diff / (60 * 60 * 1000) % 24;
		// long diffDays = diff / (24 * 60 * 60 * 1000);
		// System.out.print(diffDays + " days, ");
		// System.out.print(diffHours + " hours, ");
		// System.out.print(diffMinutes + " minutes, ");
		// System.out.print(diffSeconds + " seconds.");
		return (int) diffSeconds;
	}
}

File: livingrimoire start here/LivingGrimoire Kotlin/kotlin_core/TimeGate.kt
package com.yotamarker.lgkotlinfull.LGCore

import java.util.*

class TimeGate {
    // a gate that only opens x minutes after it has been set
    private var pause = 5 //minutes to keep gate closed
    private var openedGate = addMinutesToJavaUtilDate(Date(), pause)

    constructor(minutes: Int) : super() {
        try {
            Thread.sleep(100)
        } catch (e: InterruptedException) {
            e.printStackTrace()
        }
        pause = minutes
    }

    constructor() {}

    fun openGate() {
        openedGate = Date()
    }

    val isClosed: Boolean
        get() = !openedGate.before(Date())

    fun close() {
        openedGate = addMinutesToJavaUtilDate(Date(), pause)
    }

    fun close(minutes: Int) {
        val now = Date()
        openedGate = addMinutesToJavaUtilDate(now, minutes)
    }

    private fun addMinutesToJavaUtilDate(date: Date, minutes: Int): Date {
        val calendar = Calendar.getInstance()
        calendar.time = date
        calendar.add(Calendar.MINUTE, minutes)
        return calendar.time
    }

    fun setPause(pause: Int) {
        if (pause < 60 && pause > 0) {
            this.pause = pause
        }
    }
}